### PR TITLE
Addition of custom Headers.

### PR DESCRIPTION
**Pull request by schakrad**

### CODE REVIEW

File: extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+export const apiCall = (url: string, headers: Record<string, string>) => {
```
Comment (leoluz) [inline extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts]: Extraction of logic looks good. Thank you!
My suggestion is rename this file to client.ts and put it together with the Metrics.tsx.

File: extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  return fetch(url, { headers })
+    .then((res) => res.json())
+    .then((res) => res)
+    .catch((err) => {
+      throw err;
+    });
+};
+
+export function getDashBoard({
```
Comment (leoluz) [inline extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts]: can you please add the short documentation on exported symbols?

**Review by leoluz — CHANGES_REQUESTED** [review]
Looking good. Please take a look at my 2 comments.
Thank you

File: extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts (hunk 1 above)
Comment (schakrad) [reply extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts]: Sure, Made the modification.

File: extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts (hunk 2 above)
Comment (schakrad) [reply extensions/resource-metrics/resource-metrics-extention/ui/src/utils/api.ts]: Added comments on top of each function

**Review by leoluz — APPROVED** [review]
LGTM
thank you!

**Pull request merged**